File: GoodWill_DotNet_BackEnd/GoodWillCharity_DotNet_API/Controllers/AdminController.cs
using GoodWillCharity_DotNet_API.Models;
using GoodWillCharity_DotNet_API.Services.Abstract;
using Microsoft.AspNetCore.Mvc;

// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860

namespace GoodWillCharity_DotNet_API.Controllers
{
    [Route("api/admin")]
    [ApiController]
    public class AdminController : ControllerBase
    {
        private readonly IAdminService _adminService;
        private readonly IFileService _fileService;

        public AdminController(IAdminService admin, IFileService fileService)
        {
            _adminService = admin;
            _fileService = fileService;
        }
        [HttpPost]
        public async Task<IActionResult> Post([FromForm] Admin admin)
        {
            if (admin.ImagesFile != null)
            {
                var fileResult = _fileService.SaveImage(admin.ImagesFile);
                if (fileResult.Item1 == 1)
                {
                    admin.ProfilePicPath = fileResult.Item2;
                    Console.WriteLine(admin.ProfilePicPath);
                }
            }
            admin.Role = "Admin";
            try
            {
                await _adminService.CreateAsync(admin);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                return NotFound("Error While Adding Data");

            }
            return Ok("Admin Added Into DB");
        }
        [HttpPost]
        [Route("login")]
        public async Task<IActionResult> Login(string email, string password)
        {
            try
            {
                var admin = await _adminService.LoginAdmin(email, password);
                if (admin != null)
                {
                    //TokenGenrator<Admin> tg = new TokenGenrator<>(IConfiguration cgf);
                    string tokenValue = _adminService.GenrateToken(admin);
                    string msgout = "Login Success";
                    return Ok(new { id = admin.Id, userRole = admin.Role, msg = msgout, token = tokenValue });
                }
                else
                {
                    return BadRequest("Please Enter Valid Email or Password");
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                return BadRequest("Error");
            }

        }
    }
}

File: GoodWill_Spring_BackEnd/src/main/java/com/GoodWillCharity_SprinBoot_API/app/controller/AdminController.java
////package com.GoodWillCharity_SprinBoot_API.app.controller;
////
////import com.GoodWillCharity_SprinBoot_API.app.model.Admin;
////import com.GoodWillCharity_SprinBoot_API.app.service.IAdminService;
////import com.GoodWillCharity_SprinBoot_API.app.service.IFileService;
////import org.springframework.beans.factory.annotation.Autowired;
////import org.springframework.http.ResponseEntity;
////import org.springframework.web.bind.annotation.*;
////import org.springframework.web.multipart.MultipartFile;
////
////@RestController
////@RequestMapping("/api/admin")
////public class AdminController {
////
////    @Autowired
////    private IAdminService adminService;
////
////    @Autowired
////    private IFileService fileService;
////
////    @PostMapping
////    public ResponseEntity<String> addAdmin(@RequestParam("ImagesFile") MultipartFile imagesFile,
////                                           @ModelAttribute Admin admin) {
////        try {
////            if (!imagesFile.isEmpty()) {
////                String profilePicPath = fileService.saveImage(imagesFile);
////                admin.setProfilePicPath(profilePicPath);
////            }
////            admin.setRole("Admin");
////            adminService.createAdmin(admin);
////            return ResponseEntity.ok("Admin added into DB");
////        } catch (Exception e) {
////            e.printStackTrace();
////            return ResponseEntity.status(404).body("Error while adding data");
////        }
////    }
////
////    @PostMapping("/login")
////    public ResponseEntity<?> loginAdmin(@RequestParam String email, @RequestParam String password) {
////        try {
////            Admin admin = adminService.loginAdmin(email, password);
////            if (admin != null) {
////                String tokenValue = adminService.generateToken(admin);
////                return ResponseEntity.ok(new LoginResponse(admin.getId(), admin.getRole(), "Login Success", tokenValue));
////            } else {
////                return ResponseEntity.badRequest().body("Please enter valid email or password");
////            }
////        } catch (Exception e) {
////            e.printStackTrace();
////            return ResponseEntity.badRequest().body("Error");
////        }
////    }
////
////    private static class LoginResponse {
////        private String id;
////        private String userRole;
////        private String msg;
////        private String token;
////
////        public LoginResponse(String id, String userRole, String msg, String token) {
////            this.id = id;
////            this.userRole = userRole;
////            this.msg = msg;
////            this.token = token;
////        }
////
////        // Getters and Setters
////    }
////}
//
//
//package com.GoodWillCharity_SprinBoot_API.app.controller;
//
//import com.GoodWillCharity_SprinBoot_API.app.model.Admin;
//import com.GoodWillCharity_SprinBoot_API.app.service.IAdminService;
//import com.GoodWillCharity_SprinBoot_API.app.service.IFileService;
//import org.springframework.beans.factory.annotation.Autowired;
//import org.springframework.http.ResponseEntity;
//import org.springframework.web.bind.annotation.*;
//import org.springframework.web.multipart.MultipartFile;
//
//@RestController
//@RequestMapping("/api/admin")
//public class AdminController {
//
//    @Autowired
//    private IAdminService adminService;
//
//    @Autowired
//    private IFileService fileService;
//
//    @PostMapping
//    public ResponseEntity<String> addAdmin(@RequestParam("ImagesFile") MultipartFile imagesFile,
//                                           @ModelAttribute Admin admin) {
//        try {
//            if (!imagesFile.isEmpty()) {
//                String profilePicPath = fileService.saveImage(imagesFile);
//                admin.setProfilePicPath(profilePicPath);
//            }
//            admin.setRole("Admin");
//            adminService.createAsync(admin).join();
//            return ResponseEntity.ok("Admin added into DB");
//        } catch (Exception e) {
//            e.printStackTrace();
//            return ResponseEntity.status(404).body("Error while adding data");
//        }
//    }
//
//    @PostMapping("/login")
//    public ResponseEntity<?> loginAdmin(@RequestParam String email, @RequestParam String password) {
//        try {
//            Admin admin = adminService.loginAdmin(email, password).join();
//            if (admin != null) {
//                String tokenValue = adminService.generateToken(admin);
//                return ResponseEntity.ok(new LoginResponse(admin.getId(), admin.getRole(), "Login Success", tokenValue));
//            } else {
//                return ResponseEntity.badRequest().body("Please enter valid email or password");
//            }
//        } catch (Exception e) {
//            e.printStackTrace();
//            return ResponseEntity.badRequest().body("Error");
//        }
//    }
//
//    private static class LoginResponse {
//        private String id;
//        private String userRole;
//        private String msg;
//        private String token;
//
//        public LoginResponse(String id, String userRole, String msg, String token) {
//            this.id = id;
//            this.userRole = userRole;
//            this.msg = msg;
//            this.token = token;
//        }
//
//        // Getters and Setters
//        public String getId() { return id; }
//        public void setId(String id) { this.id = id; }
//        public String getUserRole() { return userRole; }
//        public void setUserRole(String userRole) { this.userRole = userRole; }
//        public String getMsg() { return msg; }
//        public void setMsg(String msg) { this.msg = msg; }
//        public String getToken() { return token; }
//        public void setToken(String token) { this.token = token; }
//    }
//}
